components/session-service/server/server.go

```diff
@@ -579,20 +582,27 @@ func (s *Server) newHandler(w http.ResponseWriter, r *http.Request) {
 	// take state we've gotten from the client, store it
 	if clientState := r.FormValue("state"); clientState != "" {
 		s.log.Debugf("storing clientState %s", clientState)
-		if err := sess.PutString(w, clientStateKey+"_"+relayState, clientState); err != nil {
+		if err := sess.PutString(w, clientStateKeyPrefix+relayState, clientState); err != nil {
 			http.Error(w, "failed to set client state", http.StatusInternalServerError)
 			return
 		}
 	}
 
 	// bind relay state to session
-	if err := sess.PutBool(w, relayStateKey+"_"+relayState, true); err != nil {
+	if err := sess.PutBool(w, relayStateKeyPrefix+relayState, true); err != nil {
 		s.log.Errorf("couldn't put relay state into session: %s", err)
 		httpError(w, http.StatusInternalServerError)
 		return
 	}
 	s.log.Debugf("stored relayState %s", relayState)
 	authCodeURL := s.client.AuthCodeURL(relayState)
+
+	if err := sess.PutBool(w, aliveKey, true); err != nil {
```

Comment:
This way, we don't rely on having something in session data to conclude that it's a valid session; either it has `alive:true`, or it isn't.